packages/x-components/src/x-modules/empathize/components/__tests__/empathize.spec.ts

```diff
@@ -1,53 +1,110 @@
-import { mount } from '@vue/test-utils';
+import { mount, Wrapper } from '@vue/test-utils';
 import Vue from 'vue';
-import BaseEventButton from '../../../../components/base-event-button.vue';
 import { getXComponentXModuleName, isXComponent } from '../../../../components/x-component.utils';
 import { getDataTestSelector, installNewXPlugin } from '../../../../__tests__/utils';
 import Empathize from '../empathize.vue';
+import { empathizeXModule } from '../../x-module';
+import { XPlugin } from '../../../../plugins/index';
+import { XEvent } from '../../../../wiring/index';
+
+jest.useFakeTimers();
 
 describe('testing empathize component', () => {
-  let localVue: typeof Vue;
+  function renderEmpathize({
+    eventsToOpenEmpathize = ['UserClickedSearchBox'],
+    eventsToCloseEmpathize = ['UserClosedEmpathize'],
+    template = `<Empathize v-bind="$attrs">
+                  <template #default>
+                    <span data-test="empathize-content">Empathize</span>
+                  </template>
+                </Empathize>`
+  }: RenderEmpathizeOptions = {}): RenderEmpathizeAPI {
+    const [, localVue] = installNewXPlugin();
+    XPlugin.registerXModule(empathizeXModule);
 
-  beforeEach(() => {
-    [, localVue] = installNewXPlugin({});
-  });
+    const parent = document.createElement('div');
+    document.body.appendChild(parent);
+
+    const wrapper = mount(
+      {
+        components: { Empathize },
+        template
+      },
+      {
+        localVue,
+        attachTo: parent,
+        propsData: {
+          eventsToOpenEmpathize,
+          eventsToCloseEmpathize
+        }
+      }
+    ).findComponent(Empathize);
+
+    return {
+      wrapper
+    };
+  }
 
   it('is an XComponent which has an XModule', () => {
-    const empathize = mount(Empathize, { localVue });
-    expect(isXComponent(empathize.vm)).toEqual(true);
-    expect(getXComponentXModuleName(empathize.vm)).toEqual('empathize');
+    const { wrapper } = renderEmpathize();
+    expect(isXComponent(wrapper.vm)).toEqual(true);
+    expect(getXComponentXModuleName(wrapper.vm)).toEqual('empathize');
+  });
+
+  it('will not open empathize if there is no content to render', async () => {
+    const template = `<Empathize v-bind="$attrs"/>`;
+    const { wrapper } = renderEmpathize({ template });
+
+    wrapper.vm.$x.emit('UserClickedSearchBox');
+    // Fast-forward until all timers have been executed
+    jest.runAllTimers();
+    await wrapper.vm.$nextTick();
+
+    expect(wrapper.find('.x-empathize').exists()).toBe(true);
+    expect(wrapper.find(getDataTestSelector('empathize-content')).exists()).toBe(false);
   });
 
   it('listens to UserOpenedEmpathize and UserClosedEmpathize by default', async () => {
```

Comment:
Maybe we can merge the previous test with this one? Also, on a side note, I don't know why we need to use "the hacky" approach of using `jest timers`, but I think this component would fit better in the `cypress component testing` environment than here, as we are checking the existence of `HTML` elements and their visibility. You don't have to change it, but just thinking out loud.

Reply:
Hi @LuisMartinez15, when `changeOpenState ` is fired, there's a Debounce applied (although the debounce time is set to 0). The event is not fired when `Empathize` has no content, so you are right, we didn't need to use `jest timers` in that case. But in this situation I thought it was necessary, otherwise tests failed. Is that the right approach, or is there something wrong I missed?

Reply:
I didn't notice the `debounce`. I guess that's why we need to run the timers.